django/tasks/utils.py

```diff
@@ -0,0 +1,63 @@
+import inspect
+import json
+import time
+from functools import wraps
+from traceback import format_exception
+
+from django.utils.crypto import get_random_string
+
+
+def is_module_level_function(func):
+    if not inspect.isfunction(func) or inspect.isbuiltin(func):
+        return False
+
+    if "<locals>" in func.__qualname__:
+        return False
+
+    return True
+
+
+def json_normalize(obj):
+    """
+    Round-trip encode object as JSON to normalize types.
+    """
+    return json.loads(json.dumps(obj))
+
+
+def retry(*, retries=3, backoff_delay=0.1):
```

Comment:
This is currently not documented and only used in tests, I think we either need to add docs or remove this